src/neptune/api.d

```diff
@@ -0,0 +1,166 @@
+/*******************************************************************************
+
+    helper methods to access the octod/github API
+
+    Copyright:
+        Copyright (c) 2017 Sociomantic Labs GmbH. All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE.txt for details.
+
+*******************************************************************************/
+
+module neptune.api;
+
+import octod.core;
+import octod.api.repos;
+
+
+/*******************************************************************************
+
+    Returns:
+        global api object to interact with github
+
+*******************************************************************************/
+
+public ref HTTPConnection api ( )
+{
+
+    static HTTPConnection _api;
+    static bool connected = false;
+
+    if (!connected)
+    {
+        _api = HTTPConnection.connect(getConf());
+        connected = true;
+    }
+
+    return _api;
+}
+
+
+/*******************************************************************************
+
+    Returns:
+        neptune configuration. If it doesn't exists, will prompt the user for
+        the required infomation
+
+*******************************************************************************/
+
+public Configuration getConf ( )
+{
+    import neptune.cmd;
+
+    Configuration cfg;
+    cfg.dryRun = false;
+
+    try cfg.oauthToken = cmd("git config neptune.oauthtoken");
+    catch (ExitCodeException) {}
+
+    if (cfg.oauthToken.length == 0)
+        return githubSetup();
+
+    return cfg;
+}
+
+
+/*******************************************************************************
+
+    Sets up the interaction with github using an oauth token
+
+    Returns:
+        set up configuration object
+
+*******************************************************************************/
+
+private Configuration githubSetup ( )
+{
+    import neptune.cmd;
+
+    import std.stdio;
+    import std.string;
+
+    string hub_oauth;
+
+    Configuration cfg;
+    cfg.dryRun = false;
+
+    try hub_oauth = cmd("git config hub.oauthtoken");
+    catch (ExitCodeException) {}
+
+    if (hub_oauth.length > 0)
+    {
+        writefln("No oauth token was found under neptune.oauthtoken, however an hub.oauthtoken was found. ");
+        if (readYesNoResponse("Would you like to use it as neptune.oauthtoken config?"))
+        {
+            cmd("git config --global neptune.oauthtoken " ~ hub_oauth);
+            cfg.oauthToken = hub_oauth;
+            return cfg;
+        }
+    }
+
+    writefln("Setting up an oauth token for github access");
```

Comment:
How about moving the code that creates oauth token based on password/username to octod? It seems widely applicable.